### DISCUSSION TITLE
How to use resize channel or stream?

### DISCUSSION THREAD

**Discussion by zxyao145** [kubernetes-client/csharp#1162] [Q&A]
Hello, I'm try to set sshd width and height, and my code like this:

```c#
var resizeStream = demux.GetStream(ChannelIndex.Resize, ChannelIndex.Resize);
var sizeInfo = "{\"Height\":24,\"Width\":114}";
var bytes = Encoding.UTF8.GetBytes(sizeInfo);
await resizeStream.WriteAsync(bytes, CancellationToken.None);
```
And it doesn't seem to work.

**Comment by tg123** [comment]
seems good, what do you mean by `not work`?
how did you verify?

**Reply by zxyao145** [reply]
The number of output columns does not change.

**Reply by zxyao145** [reply]
I added a `resize` event to the browser's `window`. When the browser window changes, the web client will request to change the resize of sshd of the pod, but the output of the `ls` command is always three columns. I don't know whether this is normal


![image](https://user-images.githubusercontent.com/24973084/210247012-e7239393-b8de-4c87-aea7-a9ef6e271881.png)

![image](https://user-images.githubusercontent.com/24973084/210247042-a9590642-0b9e-4319-9086-7bcb7879556e.png)

**Reply by tg123** [reply]
seems not correct 
check if `tput cols` changes after resizing

**Reply by zxyao145** [reply]
> seems not correct check if `tput cols` changes after resizing

Maybe something error, `tput cols` output always is 80, even I hard coded to resize. 

Is there a way to display the debug log?

**Reply by tg123** [reply]
here is the code https://github.com/kubernetes/kubernetes/blob/88c952d256b7e724a3cc0e6654680fda2c953c39/pkg/kubelet/cri/streaming/remotecommand/httpstream.go#L412

seems you do not have a way to track it, but you may want to build your own kubelet to see if the resizing message was sent correctly
or 
do some package capture

**Reply by zxyao145** [reply]
Get it, thanks a lot!
